src/ripple/app/tests/JTx.cpp

```diff
@@ -0,0 +1,400 @@
+    }
+    return jv;
+}
+
+Json::Value
+signers (Account const& account, none_t)
+{
+    Json::Value jv;
+    jv[jss::Account] = account.human();
+    jv[jss::TransactionType] = "SignerListSet";
+    return jv;
+}
+
+Json::Value
+trust (Account const& account,
+    STAmount const& amount)
+{
+    if (isXRP(amount))
+        throw std::runtime_error(
+            "trust() requires IOU");
+    Json::Value jv;
+    jv[jss::Account] = account.human();
+    jv[jss::LimitAmount] = amount.getJson(0);
+    jv[jss::TransactionType] = "TrustSet";
+    jv[jss::Flags] = 0;    // tfClearNoRipple;
+    return jv;
+}
+
+void
+fill_fee (Json::Value& jv,
+    Ledger const& ledger)
+{
+    if (jv.isMember(jss::Fee))
+        return;
+    jv[jss::Fee] = std::to_string(
+        ledger.getBaseFee());
+}
+
+void
+fill_seq (Json::Value& jv,
+    Ledger const& ledger)
+{
+    if (jv.isMember(jss::Sequence))
+        return;
+    RippleAddress ra;
+    ra.setAccountID(jv[jss::Account].asString());
+    auto const ar = ledger.getAccountRoot(
+        ra.getAccountID());
+    jv[jss::Sequence] =
+        ar->getFieldU32(sfSequence);
+}
+
+void
+sign (Json::Value& jv,
+    Account const& account)
+{
+    jv[jss::SigningPubKey] =
+        strHex(make_Slice(
+            account.pk().getAccountPublic()));
+    Serializer ss;
+    ss.add32 (HashPrefix::txSign);
+    parse(jv).add(ss);
+    jv[jss::TxnSignature] = strHex(make_Slice(
+        account.sk().accountPrivateSign(
+            ss.getData())));
+}
+
+
+STObject
+parse (Json::Value const& jv)
+{
+    STParsedJSONObject p("tx_json", jv);
+    if (! p.object)
+        throw parse_error(
+            rpcErrorString(p.error));
+    return std::move(*p.object);
+}
+
+void
+fee::operator()(Env const&, JTx& tx) const
+{
+    if (b_)
+        tx.fill_fee = true;
+    else if(! b_)
+        tx.fill_fee = false;
+    else if (boost::indeterminate(b_))
+        tx[jss::Fee] =
+            v_.getJson(0);
+}
+
+void
+flags::operator()(Env const&, JTx& tx) const
+{
+    tx[jss::Flags] =
+        v_ /*| tfUniversal*/;
+}
+
+void
+path::operator()(Env const& env, JTx& jtx) const
+{
+    auto& jv = jtx.jv;
+    auto const from = env.lookup(
+        jv[jss::Account].asString());
+    auto const to = env.lookup(
+        jv[jss::Destination].asString());
+    jv[jss::SendMax] = sendmax_.getJson(0);
+    STPath fp;
+    STPathSet ps;
+    auto const found = findPathsForOneIssuer(
+        std::make_shared<RippleLineCache>(
+            env.ledger), from, to,
+                sendmax_.issue(), sendmax_,
+                    depth_, limit_,
+                        ps, fp);
+    // VFALCO TODO API to allow caller to examine the STPathSet
+    // VFALCO isDefault should be renamed to empty()
+    if (found && ! ps.isDefault())
+        jv[jss::Paths] = ps.getJson(0);
+}
+
+void
+msig::operator()(Env const& env, JTx& tx) const
+{
+    // VFALCO Inefficient pre-C++14
+    auto accounts = std::move(accounts_);
```

Comment:
So `invoke` and friends take their functions as forwarding refs and we `static_assert(is_rvalue_ref(f))`? I'd like to static assert because these functlets are "run once" if we move inside of `operator()`.